### PR TITLE
Fix/zqs 925/athena caesura/mypy config

### PR DESCRIPTION
**Pull request by AthenaCaesura**
qe-qiskit now allows for mypy integration with imported repos.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -36,4 +41,6 @@
         "z-quantum-core",
     ],
     extras_require=extras,
+    package_data={"src/python": ["py.typed"]},
```
Comment (mstechly) [inline setup.py:44]: [question] It seems to me that we don't have this line in other repos (e.g. [z-quantum-core](https://github.com/zapatacomputing/z-quantum-core/blob/6667e2d969e4fa663717a2cf71ccb536bfd18678/setup.py#L60)), why it's needed here?

Comment (mstechly) [reply setup.py:44]: I see it's also in `qe-qulacs` though, I missed it in that PR.

File: subtrees/z_quantum_actions/pyproject.toml [hunk 2]
```diff
@@ -28,7 +37,10 @@ exclude_lines = [
 
 [tool.pytest.ini_options]
 log_level="INFO"
+<<<<<<< HEAD
```
Comment (mstechly) [inline subtrees/z_quantum_actions/pyproject.toml:40]: Something went wrong here! Please remove this and double-check whether it's not an issue in other places as well!

**Review by mstechly — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (jamesclark-Zapata) [reply setup.py:44]: There's a template https://github.com/zapatacomputing/z-quantum-actions/blob/main/MANIFEST.in that can be used.
This requires `include_package_data=True,` to be used to use the `MANIFEST.in`.
The method in this commit works, it just depends on the divergence from `z-quantum-actions` and how to reconcile differences in the future.

Comment (mstechly) [reply setup.py:44]: If it's needed why we don't have this in the sample setup.py in `z-quantum-actions` and why it's not in `z-quantum-core`?

Comment (jamesclark-Zapata) [reply setup.py:44]: `include_package_data=True` is in the `sample_setup.py` with a sample `MANIFEST.in` but it looks like the instructions in the ticket advocates the approach Athena used:

> add an empty py.typed file under `src/python/orquestra/<project_name>/py.typed` and add `package_data` entry to setup.py (see [this example PR](https://github.com/canonical/operator/pull/644/files))

I think our wires got crossed when writing the instructions. I don't know why `z-quantum-core` works! There's a `MANIFEST.in` but it doesn't include the `py.typed`.

Comment (AthenaCaesura) [reply setup.py:44]: Yes, I was following the instructions as James said. Although that step is perhaps unnecessary?
I think @max-radin made the `z-quantum-core` PR so maybe he would know why it worked there and why this step might required here?

File: subtrees/z_quantum_actions/pyproject.toml (hunk 2 above)
Comment (AthenaCaesura) [reply subtrees/z_quantum_actions/pyproject.toml:40]: Is this something VSCode does with merget conflicts? I don't think I've seen this anywhere else.

File: setup.py (hunk 1 above)
Comment (mstechly) [reply setup.py:44]: Ok, I'd suggest syncing with @max-radin to make sure which version is the correct one, so it doesn't backfire 3 months from now, that we have different settings in different repos.

**Review by mstechly — APPROVED** [review]

**Pull request merged**